pilot/pkg/networking/core/v1alpha3/httproute.go

```diff
@@ -137,10 +153,110 @@ func (configgen *ConfigGeneratorImpl) buildSidecarOutboundHTTPRouteConfig(env *m
 		}
 	}
 
+	cacheHit := false
+	if useSniffing && listenerPort != 0 {
+		// Check if we have already computed the list of all virtual hosts for this port
+		// If so, then  we simply have to return only the relevant virtual hosts for
+		// this listener's host:port
+		if vhosts, exists := vHostCache[listenerPort]; exists {
+			virtualHosts = getVirtualHostsForSniffedServicePort(vhosts, routeName)
+			cacheHit = true
+		}
+	}
+	if !cacheHit {
+		virtualHosts = configgen.buildSidecarOutboundVirtualHosts(env, node, push, routeName, listenerPort)
+		if listenerPort > 0 {
+			// only cache for tcp ports and not for uds
+			vHostCache[listenerPort] = virtualHosts
+		}
+
+		// FIXME: This will ignore virtual services with hostnames that do not match any service in the registry
```

Comment:
I thoguht this was fixed?

Reply:
Yes. sorry I will remove in subsequent commit. left over comment